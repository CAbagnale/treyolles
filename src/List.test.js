import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

//this is the test case

describe('List component', () => {

    const testCards = [
		{ id: 'a', title: 'first card', content: 'first card content' },
		{ id: 'b', title: 'second card', content: 'second card content' },
		{ id: 'c', title: 'third card', content: 'third card content' }
	];


    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List cards={testCards} />, div);
        ReactDOM.unmountComponentAtNode(div);
      });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(
                
                <section className="List">
                    <header className="List-header">
                        <h2>header={'testHeader'}</h2>
                    </header>
                    <div className="List-cards">
						<List cards={testCards} />
						<button type="button" className="List-add-button">
							+ Add Random Card
						</button>
					</div>
                </section>

            )
            .toJSON();
        expect(tree).toMatchSnapshot();     
    });

});
