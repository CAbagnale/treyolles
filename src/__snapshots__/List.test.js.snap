// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      header=
      testHeader
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <section
      className="List"
    >
      <header
        className="List-header"
      >
        <h2 />
      </header>
      <div
        className="List-cards"
      >
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            first card
          </h3>
          <p>
            first card content
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            second card
          </h3>
          <p>
            second card content
          </p>
        </div>
        <div
          className="Card"
        >
          <button
            type="button"
          >
            delete
          </button>
          <h3>
            third card
          </h3>
          <p>
            third card content
          </p>
        </div>
        <button
          className="List-add-button"
          type="button"
        >
          + Add Random Card
        </button>
      </div>
    </section>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
